app/models/security_scheme.rb

```diff
@@ -22,14 +41,49 @@ def parameters=(val)
     super(val)
   end
 
-  def parameters_is_valid_json
-    JSON.parse(parameters) if parameters.is_a? String
+  def valid_type_option
+    errors.add(:security_scheme_type, 'invalid choice') if SecurityScheme.security_scheme_type_options[security_scheme_type].nil?
+  end
+
+  def valid_in_option
+    return if security_scheme_in.blank?
+    errors.add(:security_scheme_in, 'invalid choice') if SecurityScheme.security_scheme_in_options[security_scheme_in].nil?
+  end
```

Comment:
Workaround.
I failed to direct use the enum. When doing so, the value in ruby was always nil.